zookeeper-server/src/main/java/org/apache/zookeeper/server/FinalRequestProcessor.java

```diff
@@ -574,13 +576,15 @@ public void processRequest(Request request) {
                         request.authInfo, path,
                         null);
                 final int maxReturned = getChildrenPaginatedRequest.getMaxReturned();
-                List<PathWithStat> list = zks.getZKDatabase().getPaginatedChildren(
+                final PaginationNextPage nextPage = new PaginationNextPage();
+                List<String> list = zks.getZKDatabase().getPaginatedChildren(
```

Comment:
If this list is ordered, you can just parse the returned list to fill next page object, right? Then you don't need to change the ZkDatabase class.

Reply:
As discussed, this is more about code style. I agree with the current layout in general.